server/src/main/java/org/elasticsearch/plugins/SearchPlugin.java

```diff
@@ -117,8 +117,9 @@
     /**
      * Allows plugins to register new aggregations using aggregation names that are already defined
      * in Core, as long as the new aggregations target different ValuesSourceTypes
+     * @return A list of the new registrar functions
      */
-    default List<Consumer<ValuesSourceRegistry>> getBareAggregatorRegistrar() {
+    default List<Consumer<ValuesSourceRegistry.ValuesSourceRegistryBuilder>> getBareAggregatorRegistrar() {
```

Comment:
`ValuesSourceRegistry.Builder`?

Reply:
Rename method to `extendAggregations` or something? Also, now that I'm looking at it, why return a list instead of just a method? Or even just:
```
default void extendAggregations(ValuesSourceRegistry.Builder builder) {}
```
?

I know it doesn't line up with the other "return lists of stuff" methods on these classes but `ValuesSourceRegistry` is a different animal anyway.